cluster/storage-cluster.go

```diff
@@ -571,3 +612,54 @@ func streamTenantService(maxChanSize int) chan TenantServiceResult {
 	}()
 	return ch
 }
+
+func SchedulePreProvisionTenantInStorageGroup(ctx *Context, sg *StorageGroup) error {
+	// first check if we can fit 1 more tenant
+	total, err := totalNumberOfTenantInStorageGroup(&sg.ID)
+	if err != nil {
+		return err
+	}
+	if total >= maxNumberOfTenantsPerSg {
+		return errors.New("Max number of tenants on that storage group reached")
+	}
+
+	// pre-provision the first tenant of this storage group
+	provTenantName := strings.ToLower(RandomCharString(16))
```

Comment:
Seems like now the tenant shortname is a random identifier, should we consider to migrate this field to uuid?

Reply:
uuid might not be url compatible, and could easy max out the max length of kubernetes names for things, so I think it's healthy to have a random character string that we control the size of

Reply:
Perhaps uuid is better